vault/src/main.rs

```diff
@@ -13,9 +20,82 @@ use vault::{
     VaultService, VaultServiceConfig, ABOUT, AUTHORS, NAME, VERSION,
 };
 
+#[derive(Parser)]
+#[clap(args_conflicts_with_subcommands = true)]
+struct Cli {
+    #[clap(subcommand)]
+    sub: Option<Commands>,
+
+    #[clap(flatten)]
+    opts: RunVaultOpts,
+}
+
+#[derive(Parser)]
+enum Commands {
+    GenerateWif(GenerateWifOpts),
+    GenerateKey(GenerateKeyOpts),
+    #[clap(name = "run")]
+    RunVault(RunVaultOpts),
+}
+
+#[derive(Debug, Parser, Clone)]
+struct GenerateWifOpts {
+    /// Output file name or stdout if unspecified.
+    #[clap(parse(from_os_str))]
+    output: Option<PathBuf>,
+
+    #[clap(long)]
+    network: Network,
+}
+
+impl GenerateWifOpts {
+    fn generate_and_write(&self) -> Result<(), Error> {
+        let secret_key = SecretKey::new(&mut thread_rng());
+        let private_key = PrivateKey::new(secret_key, self.network);
+        let wif = private_key.to_wif();
+        let data = wif.as_bytes();
+
+        if let Some(output) = &self.output {
+            std::fs::write(output, data)?;
+        } else {
+            std::io::stdout().write_all(&data)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser, Clone)]
+struct GenerateKeyOpts {
+    /// Output file name or stdout if unspecified.
+    #[clap(parse(from_os_str))]
+    output: Option<PathBuf>,
+}
+
+impl GenerateKeyOpts {
+    fn generate_and_write(&self) -> Result<(), Error> {
+        let (pair, seed) = KeyPair::generate();
```

Comment:
Is this what subkey uses?

Reply:
They seed the [`Mnemonic`](https://github.com/paritytech/substrate/blob/d65f4bbdf1d34ae228fbad444951176278a03d05/client/cli/src/commands/generate.rs#L62) using [random bytes](https://github.com/maciejhirsz/tiny-bip39/blob/4bde6ea3506d893927a371d6a92bebdce6da17e0/src/mnemonic.rs#L65) from `thread_rng()`.

Reply:
Ok, just wanted to make sure that we don't accidentally generate insecure keys